Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -2588,36 +2641,32 @@ extension SourceKitLSPServer {
 
     // Convert occurrences to type hierarchy items
     let occurs = baseOccurs + retroactiveConformanceOccurs
-    let types = occurs.compactMap { occurrence -> TypeHierarchyItem? in
+    let types = await occurs.asyncCompactMap { occurrence -> TypeHierarchyItem? in
       // Resolve the supertype's definition to find its location
       guard let definition = index.primaryDefinitionOrDeclarationOccurrence(ofUSR: occurrence.symbol.usr) else {
         return nil
       }
 
-      return indexToLSPTypeHierarchyItem2(
-        definition: definition,
-        moduleName: definition.location.moduleName,
-        index: index
-      )
+      let moduleName = definition.location.moduleName.isEmpty ? nil : definition.location.moduleName
```

Comment:
Why is the change for `moduleName.isEmpty` necessary here? I agree that it might be good to add but if so, it should be part of a separate PR. Just in case issues arise, it makes bisecting easier if each PR only addresses one issue. Same for the other checks for empty modules.